src/transformStateWithClones.js

```diff
@@ -7,7 +7,52 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  const history = [];
+  let previousState = { ...state };
+
+  for (const action of actions) {
+    switch (action.type) {
+      case 'addProperties':
+        previousState = createStateWithAddedProperties(
+          previousState,
+          action.extraData,
```

Comment:
The function name `createStateWithAddedProperties` is not descriptive of the fact that it creates a copy of the state with added properties. Consider renaming it to something like `copyStateWithAddedProperties` to clearly indicate its behavior.